File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/domain/entity/promotion/Promotion.kt
package com.sudo248.sudoo.domain.entity.promotion

import com.sudo248.base_android.base.ItemDiff

data class Promotion(
    val promotionId: String,
    val supplierId: String?,
    val value: Double,
    val name: String,
    val enable: Boolean,
    val image: String,
    val totalAmount: Int,
) : ItemDiff, java.io.Serializable {
    override fun isContentTheSame(other: ItemDiff): Boolean {
        return other is Promotion && other == this
    }

    override fun isItemTheSame(other: ItemDiff): Boolean {
        return other is Promotion && other.promotionId == promotionId && other.enable == enable
    }

}

File: client/web/sudoo/lib/domain/model/promotion/promotion.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sudoo/data/api/base_request.dart';

part 'promotion.g.dart';

@JsonSerializable()
class Promotion implements BaseRequest {
  String promotionId;
  String? supplierId;
  String name;
  double value;
  String image;
  bool? enable;
  int totalAmount;

  Promotion(this.promotionId, this.supplierId, this.name, this.value, this.image, this.totalAmount, [this.enable]);

  factory Promotion.empty() {
    return Promotion("", null, "", 0, "",0);
  }

  factory Promotion.fromJson(Map<String, dynamic> json) => _$PromotionFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$PromotionToJson(this);
}
